### ISSUE TITLE
Porting this app to the modern era

### ISSUE THREAD

**Issue by S0ulf3re** [adium/adium#48]
This app is basically dead. Which is really a shame because it's a nice one and I like having a classic styled IM client for doing stuff with. It seems to have really fallen behind, looking at how must of it is still using Objective-C. Will it ever be ported to Apple Silicon and given a bit of rejuvenation? I could try to help even though I don't know Objective-C, I have been working with Swift for a while.

**Comment by fuzzy76** [comment]
There are so many new macOS libraries now that porting doesn't really make any sense. You'd probably be better off starting with libpurple alone and build a brand new client on top.

**Comment by codeandcreate** [comment]
Nevertheless i would prefer to have adium also native on apple silicon, @fuzzy76. a compile seems to be blocked by for some libs not configured for arm64 atm (libpurple for example). Maybe there should be a bounty for porting? ^^

**Comment by codeandcreate** [comment]
i created a bounty for this "issue"... :) 
https://app.bountysource.com/issues/110273157-porting-this-app-to-the-modern-era

**Comment by S0ulf3re** [comment]
Really a shame. This is such a nice app, And it’s the only one I can find that has an interface I like too. Oh well, Maybe I could try. I hope Swift has a library for libpurple

**Comment by Lisias** [comment]
If by porting this to the modern era ends up with something like Telegram, Session or Teams, please don't. Let Adium rest as a tribute of a forgotten era where bloatware wasn't the norm.

* Telegram uses about 600MB of ram.
* Session uses somewhat less, 200MB in multiple processes, but I have only a few friends on there, so the milage will probably vary.
* Microsoft Teams is an abhorrent piece of digital excrement, and the damned thing eats easily 800MB to 1GB of memory, adding insult to the injury.

The only laudable mention is:
* Trillian is **excellent**, less than 50MB, but the thing is closed source and only supports XMPP and OLARK.
* And, of course, Pidgin - being the MacOS recommended client… Adium :)

So, nope. By all means, I'm already fed up with bloated software implemented as they were a web application - with us getting **the worst** of both worlds. Just no.

I would really love having Adium updated, but not at such cost.

**Comment by codeandcreate** [comment]
Yeah, it shouldn't be a "complete rewrite" and making something like Telegram or Teams out of it. It should be the same as the lastest release but recompiled for Apple Silicon and maybe with something like dark mode support... but Adium is good as it is generally!

**Comment by dijit** [comment]
Added $50 to the bounty.

Libpurple for Teams plugin sparked my interest in having a lightweight Teams client again and I remembered Adium from the early 2010's, but I'm on an M2 Mac, and would prefer it to be native. <3

**Comment by fuzzy76** [comment]
@Lisias and @codeandcreate , noone has suggested something like that. Adium has always been a native app. And that is doable with a rewrite as well. There is no need to assume Electron.

**Comment by khurshid-alam** [comment]
It is surprising to see it is not being ported for MacBook Air m1, m2. What are the alternatives for native client? I couldn’t really find any. 

Btw, can someone tell me how can we run this ( + all default libpurple plugin ) with rosetta on m1 ?

**Comment by codeandcreate** [comment]
@khurshid-alam it works for me out of the box on my m1 pro mbp

**Comment by fuzzy76** [comment]
FWIW, [Pidgin from Homebrew](https://formulae.brew.sh/formula/pidgin) works, it's just butt ugly.

Other alternatives are [All-in-One Messenger](https://allinone.im) and [Ferdium](https://ferdium.org).

**Comment by khurshid-alam** [comment]
@codeandcreate Where did you get arm64 build for air m1 ? I can only see intel build ? Or how to install it ?

**Comment by codeandcreate** [comment]
@khurshid-alam There isn't a arm64 build. the intel build works out of the box via rosetta - just drag&drop from the dmg

**Comment by janosrusiczki** [comment]
Hello, I've been reminiscing about old Mac apps and wandered around here... Be warned that bountysource seems to have pulled a scam, they are unreachable since last year. :sad: Such is the web landscape nowadays unfortunately.

**Comment by dobkun** [comment]
don't change the icons, and the ui in general though.
i prefer the aqua design

**Comment by S0ulf3re** [comment]
I know this isn’t the best idea given the controversy around it, but given how much it’s evolved, and the relatively interoperable nature of swift and objective C. I’m curious if agentic coding tools like Claude Code would be able to aid in migrating Adium. 

It obviously isn’t the best solution, but given just how dated the code is, it may be the only option for a small group of individuals.

**Comment by KnutMann** [comment]
Please have a look at my fork at KnutMann/adium. I tried to revive it with the help of AI and would like to get some feedback, especially from some long term users.

**Comment by robotive** [comment]
AI is a useful tool these days and yes it got better over the last couple of years, but it still needs an experienced engineer to do something more than a working prototype.

As for the devs of Adium, we all got older… which often times means we've got less free time, got family, kids or just moved on to other things. The thing is, with people using all kinds of proprietary messaging apps these days, there is less of a need for apps like Adium unfortunately. I am curious as to what would people use Adium these days for?

**Comment by srjfoo** [comment]
> I am curious as to what would people use Adium these days for?

I think someone mentioned in another issue that reading back through old logs would be useful, and I agree. I _**can**_ read them and suss what I need out of them, but it's a slog.

But for me, what I'd use it for is jabber, unless it also worked for gchat. Of all the interfaces I've used, I still prefer Adium. The specialized apps lack the display flexibility Adium had.

(But yes, I totally understand the "we all got older..." and things change. Thank you!)

**Comment by RobFog** [comment]
So nice to read something from you after all these years, @robotive. 😊 I hope life is treating you well.

> […] The thing is, with people using all kinds of proprietary messaging apps these days, there is less of a need for apps like Adium unfortunately. I am curious as to what would people use Adium these days for?

I actually think the opposite is true. With messaging being so fragmented across different services, I think many people would like to be able to use them all in one place. There are libpurple plugins for many of the (proprietary) messaging services (see https://pidgin.im/plugins/) which could be ported to Adium.

Libpurple (and Pidgin) is still very much in development (see recent commits at https://keep.imfreedom.org/pidgin/pidgin/), with Pidgin 3 being in alpha stage (see recent release announcements at https://discourse.imfreedom.org/c/pidgin/5). Libpurple 3 is apparently designed to support modern messaging features.

**Comment by robotive** [comment]
> So nice to read something from you after all these years, [@robotive](https://github.com/robotive). 😊 I hope life is treating you well.
> 

Hey @RobFog, nice to hear from you too, really sorry we didn't stay in touch actually…  eventually i've figured i am not very good when it comes to people 😅.
Life is fine, complicated but fine… although this is nothing new i guess. Hope you are well too!

> > […] The thing is, with people using all kinds of proprietary messaging apps these days, there is less of a need for apps like Adium unfortunately. I am curious as to what would people use Adium these days for?
> 
> I actually think the opposite is true. With messaging being so fragmented across different services, I think many people would like to be able to use them all in one place. There are libpurple plugins for many of the (proprietary) messaging services (see https://pidgin.im/plugins/) which could be ported to Adium.
> 
> Libpurple (and Pidgin) is still very much in development (see recent commits at https://keep.imfreedom.org/pidgin/pidgin/), with Pidgin 3 being in alpha stage (see recent release announcements at https://discourse.imfreedom.org/c/pidgin/5). Libpurple 3 is apparently designed to support modern messaging features.

To be honest i am surprised that there is so much going on with pidgin still, especially when it comes to proprietary protocols… although i didn't looked into it in years.

Funny thing is i still have the Adium in my Applications folder, even though i didn't run it in years and don't even have Rosetta installed.

I am curious as to which of the branches has the latest state? And there also used to be a 1.6 branch that we never released in the end iirc.

**Comment by KnutMann** [comment]
OK, maybe I didn't express myself very well when I said "tried to revive“. This is not a prototype anymore. The fork is usable day to day, with only a small set of known issues left.

It includes a substantial ARC migration, a new preferences system replacing the old preference NIBs, migration from Growl to Notification Center, current Xcode/Apple Silicon/macOS compatibility work, updated dependencies and protocol plug-ins.

I've used Adium for decades, so thank you for all the work that made it such a great application in the first place.

**Comment by RobFog** [comment]
> Hey [@RobFog](https://github.com/RobFog), nice to hear from you too, really sorry we didn't stay in touch actually… eventually i've figured i am not very good when it comes to people 😅. Life is fine, complicated but fine… although this is nothing new i guess. Hope you are well too!

Heh. Same here!

> To be honest i am surprised that there is so much going on with pidgin still, especially when it comes to proprietary protocols… although i didn't looked into it in years.

Indeed. I guess it shows there's a demand. :-)

> Funny thing is i still have the Adium in my Applications folder, even though i didn't run it in years and don't even have Rosetta installed.

Yep, me too.

> I am curious as to which of the branches has the latest state? And there also used to be a 1.6 branch that we never released in the end iirc.

1.7hg was quite unstable. The 1.6 branch is more stable, perhaps alpha or beta quality, and has some nice improvements over the most recent 1.5.x release.

**Comment by fuzzy76** [comment]
Here is a clickable link to the fork: https://github.com/KnutMann/adium
I already wish for a issuetracker and possibly a build

**Comment by robotive** [comment]
@KnutMann Any chance you moved over features from 1.6 branch into your fork?

**Comment by KnutMann** [comment]
@robotive

I found code in other repos:
- Adium 1.6:
    jas8522/adium – adium-1.6 (https://github.com/jas8522/adium/tree/adium-1.6)
    shtrom/adium – adium-1.6 (https://github.com/shtrom/adium/tree/adium-1.6)
- Adium 1.7hg:
    jas8522/adium – master (https://github.com/jas8522/adium)
    shtrom/adium – master (https://github.com/shtrom/adium)

I cherry picked from there. Some things were already done when I found the repos, some things are still planned. 
Anything in particular that I should have a look at?

**Comment by RobFog** [comment]
One of 1.6's major changes was a reorganisation of Adium's preferences. Screenshot from 1.6hg: 

<img width="712" height="587" alt="Image" src="https://github.com/user-attachments/assets/c446cbb1-839b-4146-90d0-1f529ce27155" />

Does your fork look like that, @KnutMann?

**Comment by KnutMann** [comment]
<img width="982" height="760" alt="Image" src="https://github.com/user-attachments/assets/79af5f83-febc-4e78-ac07-c533735a7d1b" />

I redesigned the entire settings window to match the style of macOS. I got rid of the 27 different language nibs for all the preference panes that way. I used the high res icons that came with 1.6.

**Comment by RobFog** [comment]
Oh, wow. That looks neat.

**Comment by robotive** [comment]
> [@robotive](https://github.com/robotive)
> 
> I found code in other repos:
> 
> * Adium 1.6:
>   jas8522/adium – adium-1.6 (https://github.com/jas8522/adium/tree/adium-1.6)
>   shtrom/adium – adium-1.6 (https://github.com/shtrom/adium/tree/adium-1.6)
> * Adium 1.7hg:
>   jas8522/adium – master (https://github.com/jas8522/adium)
>   shtrom/adium – master (https://github.com/shtrom/adium)
> 
> I cherry picked from there. Some things were already done when I found the repos, some things are still planned. Anything in particular that I should have a look at?

I don't have anything in particular in mind given that time has passed, but i know that quite some work was put on that branch and it would've been nice to get all the things that never made it into main/master branch, so we don't loose those improvements.

**Comment by fuzzy76** [comment]
Ideally, KnutMann should be let aboard as a maintainer of the original.

**Comment by dijit** [comment]
due to Jia Tan shaped issues, I am against that proposal.

No offence.

**Comment by ChrisForsythe** [comment]
> due to Jia Tan shaped issues, I am against that proposal.
> 
> No offence.

Please clarify which part, there's a ton of comments on this thing.

**Comment by dijit** [comment]
> > due to Jia Tan shaped issues, I am against that proposal.
> > 
> > No offence.
> 
> Please clarify which part, there's a ton of comments on this thing. 

Sorry, you're quite right.




I meant this:

> Ideally, KnutMann should be let aboard as a maintainer of the original.

**Comment by ChrisForsythe** [comment]
> > > due to Jia Tan shaped issues, I am against that proposal.
> > > 
> > > No offence.
> > 
> > Please clarify which part, there's a ton of comments on this thing. 
> 
> Sorry, you're quite right.
> 
> 
> 
> 
> I meant this:
> 
> > Ideally, KnutMann should be let aboard as a maintainer of the original.
> 
> 

No worries, I figured but it's good to write it out so people can follow along. 

For those unfamiliar, the "Jia Tan" issue is that a random contributor was promoted to push code to a main repo, put in a back door, and was caught as a performance regression. 

I would say it's unlikely but could be possible that it happens here. However, I would also say that there are some facts here:

1) The main repo really hasn't had activity in 6 years. 

2) Contributing to the repo really doesn't push an updated binary to users. 

3) Code review exists


A way to balance this would for there to be PR's reviewed before merging. However, who is left to review them? 

This project has always been people who are semi anonymous to anonymous, working on something fun, and coming and going as they pleased. I had life going on and 10k tickets in trac TK manage years ago and it was too much for me personally. I think a lot of people had similar things happen. 

If someone is showing interest, my vote is for this

1) give them repo access after a few merged revisions

2) revert their changes if bad stuff happens

3) find someone the project trusts to commit to a once every x amount of time review period. It can be one or more past contributors. 

4) let people know what this cadence is. 


If no one volunteers to review like this, then saying no to granting the rights to commit directly effectively drives the last nail into the coffin. 

Adium wouldn't exist if this is how it was handled, at least in this kind of iteration. Neither would Growl or Perian. 



The nice thing with a public vcs like this is that stuff can be reverted. Adium has always been a learning to code on the side kind of project to a lot of people, I think suddenly saying no because of one instance (or a few) on another operating system isn't worth worrying about as much. 

I've typed a lot. I want to give others a chance to provide their own opinions. I don't really have a vote here, just a thought process and some back history of how the project has been run.

**Comment by KnutMann** [comment]
> No offence.

None taken from my side. My fork was meant for myself to keep it running and along the way I thought I had done too much just to let it go to waste. Access to the original repo isn't something I had in mind.

**Comment by robotive** [comment]
Iirc i have done a few PRs/patches before i have been granted commit access and to me it made sense that way. Regardless of that, i think having PRs before merging, as opposed to just merging directly is still a good idea.

I can't commit to anything or promise, etc. but theoretically i could spare some of my limited free time in the near future and do some reviews. Just probably not in the coming 1-2 weeks, since i am in the middle of an important release scheduled for the beginning of September.

With that said, reviewing and merging anything into Adium these days would only make sense to me if we'd like to publish a release afterwards.
